Remotion/Web/UnitTests/PairConstraint.cs

```diff
@@ -23,42 +23,30 @@ namespace Remotion.Web.UnitTests
   public class PairConstraint : EqualConstraint
   {
     private readonly Pair _expected;
-    private Pair _actual;
 
     public PairConstraint (Pair expected)
-        : base(expected)
+        : base (expected)
     {
       _expected = expected;
     }
 
-    public override bool Matches (object actual)
+    public override ConstraintResult ApplyTo<TActual> (TActual actual)
     {
-      _actual = actual as Pair;
+      var actualPair = actual as Pair;
+      var baseResult = base.ApplyTo (actual);
+      if (baseResult.IsSuccess)
+        return baseResult;
 
-      if (base.Matches (actual))
-        return true;
+      return new PairConstraintResult (this, actualPair, _expected, Matches (actualPair));

```

Comment:
I think we should move the Matches into an extra statement
```
var isSuccess  = Matches (actualPair);
```

Otherwise, we have too much going on in the single statement.